### ISSUE TITLE
Compilation fails on OS X 10.10

### ISSUE THREAD

**Issue by jhead** [addaleax/lzma-native#6]
`npm install lzma-native` fails with the following error.

io.js v2.4.0
node-gyp v2.0.1

```
...
Making install in tests
make[3]: Nothing to be done for `install-exec-am'.
make[3]: Nothing to be done for `install-data-am'.
make[3]: Nothing to be done for `install-exec-am'.
 xz-5.2.1/build-aux/install-sh -c -d '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz'
 /usr/bin/install -c -m 644 xz-5.2.1/AUTHORS xz-5.2.1/COPYING xz-5.2.1/COPYING.GPLv2 xz-5.2.1/NEWS xz-5.2.1/README xz-5.2.1/THANKS xz-5.2.1/TODO xz-5.2.1/doc/faq.txt xz-5.2.1/doc/history.txt xz-5.2.1/doc/xz-file-format.txt xz-5.2.1/doc/lzma-file-format.txt '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz'
 xz-5.2.1/build-aux/install-sh -c -d '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz/examples'
 /usr/bin/install -c -m 644 xz-5.2.1/doc/examples/00_README.txt xz-5.2.1/doc/examples/01_compress_easy.c xz-5.2.1/doc/examples/02_decompress.c xz-5.2.1/doc/examples/03_compress_custom.c xz-5.2.1/doc/examples/04_compress_easy_mt.c xz-5.2.1/doc/examples/Makefile '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz/examples'
 xz-5.2.1/build-aux/install-sh -c -d '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz/examples_old'
 /usr/bin/install -c -m 644 xz-5.2.1/doc/examples_old/xz_pipe_comp.c xz-5.2.1/doc/examples_old/xz_pipe_decomp.c '/Users/justin/<path>/node_modules/lzma-native/build/liblzma/build/share/doc/xz/examples_old'
  TOUCH Release/obj.target/liblzma.stamp
  CXX(target) Release/obj.target/lzma_native/src/util.o
  CXX(target) Release/obj.target/lzma_native/src/liblzma-functions.o
  CXX(target) Release/obj.target/lzma_native/src/filter-array.o
  CXX(target) Release/obj.target/lzma_native/src/lzma-stream.o
../src/lzma-stream.cpp:128:20: error: no member named 'move' in namespace 'std'
        self->inbufs.push(LZMA_NATIVE_MOVE(inputData));
                          ^~~~~~~~~~~~~~~~
../src/liblzma-node.hpp:39:31: note: expanded from macro 'LZMA_NATIVE_MOVE'
#define LZMA_NATIVE_MOVE std::move
                         ~~~~~^
../src/lzma-stream.cpp:209:12: error: no member named 'move' in namespace 'std'
                outbuf = LZMA_NATIVE_MOVE(outbufs.front());
                         ^~~~~~~~~~~~~~~~
../src/liblzma-node.hpp:39:31: note: expanded from macro 'LZMA_NATIVE_MOVE'
#define LZMA_NATIVE_MOVE std::move
                         ~~~~~^
../src/lzma-stream.cpp:307:13: error: no member named 'move' in namespace 'std'
                                inbuf = LZMA_NATIVE_MOVE(inbufs.front());
                                        ^~~~~~~~~~~~~~~~
../src/liblzma-node.hpp:39:31: note: expanded from macro 'LZMA_NATIVE_MOVE'
#define LZMA_NATIVE_MOVE std::move
                         ~~~~~^
../src/lzma-stream.cpp:330:13: error: no member named 'emplace' in 'std::queue<std::vector<unsigned char,
      std::allocator<unsigned char> >, std::deque<std::vector<unsigned char, std::allocator<unsigned char> >,
      std::allocator<std::vector<unsigned char, std::allocator<unsigned char> > > > >'
                                outbufs.emplace(outbuf.data(), outbuf.data() + outsz);
                                ~~~~~~~ ^
4 errors generated.
make: *** [Release/obj.target/lzma_native/src/lzma-stream.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/iojs/v2.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.1.0
gyp ERR! command "/usr/local/iojs/v2.4.0/bin/iojs" "/usr/local/iojs/v2.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/justin/<path>/node_modules/lzma-native
gyp ERR! node -v v2.4.0
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok 
npm ERR! Darwin 14.1.0
npm ERR! argv "/usr/local/iojs/v2.4.0/bin/iojs" "/usr/local/bin/npm" "install" "lzma-native"
npm ERR! node v2.4.0
npm ERR! npm  v2.13.0
npm ERR! code ELIFECYCLE

npm ERR! lzma-native@0.2.13 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the lzma-native@0.2.13 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the lzma-native package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls lzma-native
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/justin/<path>/npm-debug.log
```

**Comment by addaleax** [comment]
Thanks for the report! The errors regarding `LZMA_NATIVE_MOVE` probably come from not including `utility.h` – That is easy to fix.
`emplace` should be available in `std::queue` for compilers supporting C++11 (and, since `LZMA_NATIVE_MOVE` is used, the compiler seems to indicate support). It may be quite helpful to know your compiler and compiler version (For GCC, check `g++ --version`); Could you post that here?

**Comment by jhead** [comment]
Hello!

Here's the output as requested.

```bash
$ g++ --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.1.0
Thread model: posix
```

Let me know if you need any further info.

**Comment by jhead** [comment]
Looks like this might be related? http://svn.boost.org/trac/boost/ticket/8092

**Comment by jhead** [comment]
I was able to get it successfully compiled using the latest version of `gcc` from brew, like so.

```bash
$ brew install gcc
$ export CXX="g++-4.9"
$ export CC="gcc-4.9"
$ npm install lzma-native
```

It seems this is an issue with version of clang, which g++/gcc point to on OS X, that is bundled with Xcode.

**Comment by jhead** [comment]
I can confirm that 5558694 fixed it using the built-in clang on OS X 10.10. Thanks! :smile:

**Comment by addaleax** [comment]
Yes, your pointer to the boost issue was perfectly correct – The problem is a combination of clang and the libstdc++ version shipped with it on OS X. Using g++ will lead to a more recent version of libstdc++, so that works in any case.

Still, would you mind testing the fix I just pushed? I did not publish to npm yet, but you can test using
`git clone git@github.com:addaleax/lzma-native.git && cd lzma-native && CXX=clang++ CC=clang npm install --clang=1`

**Comment by addaleax** [comment]
Ah, sorry, did not see your comment yet – Thanks for testing!

**Comment by addaleax** [comment]
Published as 0.2.14
